libass/ass_shaper.c

```diff
@@ -234,6 +234,20 @@ get_cached_metrics(struct ass_shaper_metrics_data *metrics,
     return val;
 }
 
+size_t ass_face_size_metrics_construct(void *key, void *value, void *priv)
+{
+    FaceSizeMetricsHashKey *k = key;
+    FT_Size_Metrics *v = value;
+
+    FT_Face face = k->font->faces[k->face_index];
+
+    ass_face_set_size(face, k->size);
```

Comment:
`glyph` and `face_size` construct functions now mutating the passed in face object might warrant a doc hint?

Reply:
Hmm, reasonable enough; where do you think such a comment should go? (I generally think of the face size as transient, as we always set it and then immediately perform any operations that rely on it; if anything it was unusual that previously we were setting the size and performing operations depending on it so far apart from each other)

Reply:
The mutation ultimately happens when calling `ass_cache_get` so at one hand documenting it here would make sense — but it only affects a specific cache.
Probably best handled like the ownership docs we discussed on IRC: add a generic note and pointer to where to find each specific cache’s properties to `ass_cache_get` and then at the more specific place for the glyph and face size metrics caches note what get’s mutated etc

Reply:
Added in ass_cache.h and ass_cache_template.h.

Not going to write up the more general docs on each function's behavior right now, since that's about to be simplified substantially in the threading PR.

Reply:
yup, that’s fine; thanks!